### ISSUE TITLE
One badly defined resource (E0421) should not make traefik ingress out of service

### ISSUE THREAD

**Issue by AntoineNGUYEN31** [traefik/traefik#8088]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Install traefik 2.4.8 using helm:
```
helm install traefik-ext traefik/traefik --version 9.18.1
```

Create a middleware resource with error syntax:
```
kubectl apply -f bad_middleware.yaml
```
Create another resource with good definition, e.g. another middleware
```
kubectl apply -f good_middleware.yaml
```

### What did you expect to see?

Good resources should be handled and also shown in dashboard.
Error about the badly defined resource is shown in the pod.

Having a badly defined resource should not make the whole traefik ingress out of service.

![kuberntes-resource-is-handled-when-syntax-error-is-corrected](https://user-images.githubusercontent.com/50294775/115592336-61f24a80-a2d3-11eb-8f89-29bfa8f87b18.PNG)

### What did you see instead?

Nothing is created in dashboard and nothing is handled by traefik: ingressroute, middleware, etc.

![no-more-kubernetes-resource-can-be-created](https://user-images.githubusercontent.com/50294775/115592342-63bc0e00-a2d3-11eb-99f5-e18f6861b6c6.PNG)

Error about badly defined resource can be seen in the logs:

```
kubectl logs --tail 1 traefik-ext-mvb6h
E0421 16:34:26.196227       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Middleware: failed to list *v1alpha1.Middleware: v1alpha1.MiddlewareList.Items: []v1alpha1.Middleware: v1alpha1.Middleware.Spec: v1alpha1.MiddlewareSpec.RedirectRegex: dynamic.RedirectRegex.Permanent: ReadBool: expect t or f, but found ", error found in #10 byte of ...|rmanent":"true","reg|..., bigger context ...|54418a68e"},"spec":{"redirectRegex":{"permanent":"true","regex":"^http://localhost/(.*)","replacemen|...
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
2.4.8
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
kubernetes
```

bad_middleware.yaml
```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: bad-redirectregex
  annotations:
    kubernetes.io/ingress.class: traefik-ext
spec:
  redirectRegex:
    permanent: "true"
    regex: ^http://localhost/(.*)
    replacement: http://mydomain/${1}
```
good_middleware.yaml
```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: good-redirectregex
  annotations:
    kubernetes.io/ingress.class: traefik-ext
spec:
  redirectRegex:
    permanent: true
    regex: ^http://localhost/(.*)
    replacement: http://mydomain/${1}

```

**Comment by jbdoumenjou** [comment]
Hi @AntoineNGUYEN31,

Thanks for your interest in Traefik!
As it is a validation error, it will be addressed by the #5473
The new CRDs are available in the experimental [documentation](https://doc.traefik.io/traefik/master/reference/dynamic-configuration/kubernetes-crd/) and will be shipped in the next minor version (traefik v2.5) that should come soon :smile:

**Comment by AntoineNGUYEN31** [comment]
Hi @jbdoumenjou,

Thank you for your action. 

Just to make sure that the gravity is well captured: not only middleware resource, but also, ingressroute, ingress, etc. are not processed by traefik in this case.

FYI: the last time I upgraded the traefik ingress to 2.4.8, due to this behavior, all services exposed by ingress were not accessible.

Can't wait to see the issue fixed in v2.5 😃
